step_clone_vm.go

```diff
@@ -139,11 +142,40 @@ func cloneVM(params *CloneParameters) (vm *object.VirtualMachine, err error) {
 		datastoreRef := params.datastore.Reference()
 		relocateSpec.Datastore = &datastoreRef
 	}
+	if params.linkedClone == true {
+		relocateSpec.DiskMoveType = "createNewChildDiskBacking"
+	}
 
 	cloneSpec := types.VirtualMachineCloneSpec{
 		Location: relocateSpec,
 		PowerOn:  false,
 	}
+	if params.linkedClone == true {
+		var vmImage mo.VirtualMachine
+		err = params.vmSrc.Properties(params.ctx, params.vmSrc.Reference(), []string{"snapshot", "summary.config.template"}, &vmImage)
+		if err != nil {
+			err = fmt.Errorf("Error reading base VM properties: %s", err)
+			return
+		}
+		// TODO: is that actually true that template does not need a snapshot?
```

Comment:
base templates do need snapshots, but you cannot add it automatically because they are immutable.